File: examples/test-projects/small-projects/test-project-php/Classes.php
<?php
/** Иерархия классов — Shape и Animal (аналог classes из других малых проектов). */

class Shape {
    protected string $color;

    public function __construct(string $color = 'white') {
        $this->color = $color;
    }

    public function area(): float {
        return 0.0;
    }

    public function perimeter(): float {
        return 0.0;
    }

    public function describe(): string {
        return "Shape(color={$this->color})";
    }
}

class Circle extends Shape {
    const PI = 3.14159265;
    private float $radius;

    public function __construct(float $radius, string $color = 'white') {
        parent::__construct($color);
        $this->radius = $radius;
    }

    public function area(): float {
        return self::PI * $this->radius * $this->radius;
    }

    public function perimeter(): float {
        return 2 * self::PI * $this->radius;
    }

    public function describe(): string {
        return "Circle(r={$this->radius}, color={$this->color})";
    }
}

class Rectangle extends Shape {
    private float $width;
    private float $height;

    public function __construct(float $width, float $height, string $color = 'white') {
        parent::__construct($color);
        $this->width  = $width;
        $this->height = $height;
    }

    public function area(): float {
        return $this->width * $this->height;
    }

    public function perimeter(): float {
        return 2 * ($this->width + $this->height);
    }

    public function isSquare(): bool {
        return $this->width === $this->height;
    }

    public function describe(): string {
        return "Rectangle({$this->width}x{$this->height}, color={$this->color})";
    }
}

class Animal {
    protected string $name;
    protected string $sound;

    public function __construct(string $name, string $sound) {
        $this->name  = $name;
        $this->sound = $sound;
    }

    public function speak(): string {
        return "{$this->name} says {$this->sound}";
    }

    public function move(): string {
        return "{$this->name} moves";
    }
}

class Dog extends Animal {
    private array $tricks = [];

    public function __construct(string $name) {
        parent::__construct($name, 'Woof');
    }

    public function learnTrick(string $trick): void {
        $this->tricks[] = $trick;
    }

    public function showTricks(): string {
        if (empty($this->tricks)) {
            return "{$this->name} knows no tricks";
        }
        return "{$this->name} knows: " . implode(', ', $this->tricks);
    }
}

File: examples/test-projects/small-projects/test-project-python/classes.py
"""Иерархия классов — Shape и Animal (аналог classes из других малых проектов)."""


class Shape:
    """Базовый класс геометрической фигуры."""

    def __init__(self, color="white"):
        self.color = color

    def area(self):
        return 0.0

    def perimeter(self):
        return 0.0

    def describe(self):
        return f"Shape(color={self.color})"


class Circle(Shape):
    """Круг с радиусом."""

    PI = 3.14159265

    def __init__(self, radius, color="white"):
        super().__init__(color)
        self.radius = radius

    def area(self):
        return self.PI * self.radius * self.radius

    def perimeter(self):
        return 2 * self.PI * self.radius

    def describe(self):
        return f"Circle(r={self.radius}, color={self.color})"


class Rectangle(Shape):
    """Прямоугольник с шириной и высотой."""

    def __init__(self, width, height, color="white"):
        super().__init__(color)
        self.width = width
        self.height = height

    def area(self):
        return self.width * self.height

    def perimeter(self):
        return 2 * (self.width + self.height)

    def is_square(self):
        return self.width == self.height

    def describe(self):
        return f"Rectangle({self.width}x{self.height}, color={self.color})"


class Animal:
    """Базовый класс животного."""

    def __init__(self, name, sound):
        self.name = name
        self.sound = sound

    def speak(self):
        return f"{self.name} says {self.sound}"

    def move(self):
        return f"{self.name} moves"


class Dog(Animal):
    """Собака — подкласс Animal."""

    def __init__(self, name):
        super().__init__(name, "Woof")
        self.tricks = []

    def learn_trick(self, trick):
        self.tricks.append(trick)

    def show_tricks(self):
        if not self.tricks:
            return f"{self.name} knows no tricks"
        return f"{self.name} knows: {', '.join(self.tricks)}"
